server/src/auth/auth.controller.ts

```diff
@@ -0,0 +1,90 @@
+import { Response } from 'express'
+import { User, Prisma } from '@prisma/client'
+
+import {
+    Body,
+    Controller,
+    Get,
+    HttpCode,
+    HttpException,
+    HttpStatus,
+    Logger,
+    Post,
+    Res,
+    Req,
+    UseGuards,
+} from '@nestjs/common'
+
+import { AuthService } from './auth.service'
+import { RequestWithUser } from './dto'
+import { JwtAuthGuard, JwtAuthLocalGuard } from './guards'
+import { PostgresErrorCode } from '../../database/postgresErrorCodes.enum'
+import { UserService } from '../user/user.service'
+@Controller('auth')
+export class AuthController {
+    private readonly logger: Logger
+    constructor(
+        private readonly authService: AuthService,
+        private readonly userService: UserService,
+    ) {
+        this.logger = new Logger()
+    }
+
+    @UseGuards(JwtAuthGuard)
+    @Get()
+    authenticate(@Req() request: RequestWithUser) {
+        const user = request.user
+        user.password = undefined
+        return user
+    }
+
+    @Post('signUp')
```

Comment:
```suggestion
    @Post('signup')
```

I would rename the endpoint to "register"